src/NuGet.Services.EndToEnd/Support/Clients/RegistrationClient.cs

```diff
@@ -60,6 +59,61 @@ public RegistrationClient(SimpleHttpClient httpClient, V3IndexClient v3IndexClie
                 logger: logger);
         }
 
+        /// <summary>
+        /// Polls all registration base URLs until the provided ID and version has the specified listed state.
+        /// </summary>
+        /// <param name="id">The package ID.</param>
+        /// <param name="version">The package version.</param>
+        /// <param name="deprecation">The deprecation state to poll for.</param>
+        /// <param name="logger">The logger.</param>
+        /// <returns>Returns a task that completes when the package has the specified deprecation state or the timeout has occurred.</returns>
+        public async Task WaitForDeprecationStateAsync(
```

Comment:
IMO this polling is too complex. Perhaps just check the deprecation is non-null and let the caller do the nitty gritty

Reply:
sounds good.